src/policies/previous_guidance.rs

```diff
@@ -148,37 +134,37 @@ trait ContextIsolationGuidanceInternalExt:
-                        + symptom_record.symptom_start
-                } else {
-                    intervention_policy_parameters.mild_symptom_isolation_duration
-                        + symptom_record.symptom_start
-                };
-                let isolation_end = f64::max(minimum_isolation_time, self.get_current_time());
-                self.add_plan(isolation_end, move |context| {
-                    context
-                        .modify_itinerary_and_isolation_status(person_id, false)
-                        .unwrap();
-                    trace!("Person {person_id} is now no longer isolating");
-                    if !symptom_record.severe {
-                        context.make_post_isolation_masking_plan(
-                            person_id,
-                            intervention_policy_parameters,
-                        );
-                    }
-                });
-            }
+        if self.get_person_property(person_id, LastTestResult) {
+            // this function is called only for persons with a symptom record
+            let symptom_record = self.get_person_property(person_id, SymptomRecord).unwrap();
+            // even if there is a delay to start isolation, people's isolation time is counted
+            // from their symptom start time
+            let minimum_isolation_time = if symptom_record.severe {
+                intervention_policy_parameters.moderate_symptom_isolation_duration
+                    + symptom_record.symptom_start
+            } else {
+                intervention_policy_parameters.mild_symptom_isolation_duration
+                    + symptom_record.symptom_start
+            };
+            // schedule end of isolation in at the minimum isolation time if it has not already passed
+            // otherwise end isolation immediately
+            let isolation_end = f64::max(minimum_isolation_time, self.get_current_time());
+            self.add_plan(isolation_end, move |context| {
+                context.end_isolation(person_id).unwrap();
+                trace!("Person {person_id} is now no longer isolating");
+
+                // if the overall policy duration has not yet elapsed when isolation ends, schedule masking
+                // if moderate_symptom_isolation_duration == overall_policy_duration, this will not schedule masking
+                // for moderate cases
+                let proposed_policy_end_time = symptom_record.symptom_start
+                    + intervention_policy_parameters.overall_policy_duration;
+                if proposed_policy_end_time > context.get_current_time() {
+                    context.make_post_isolation_masking_plan(person_id, proposed_policy_end_time);
+                }
+            });
         } else {
-            self.modify_itinerary_and_isolation_status(person_id, false)
-                .unwrap();
+            // If a person's latest test is false or none, then they end isolation.
```

Comment:
```suggestion
            // If a person's latest test is false, then they end isolation.
```
I believe none is no longer an option?